code/notificationbuilder/src/main/java/com/adobe/marketing/mobile/notificationbuilder/internal/builders/TimerNotificationBuilder.kt

```diff
@@ -0,0 +1,188 @@
+ */
+internal object TimerNotificationBuilder {
+
+    data class TimerContent(
+        val title: String,
+        val body: String?,
+        val expandedBody: String?,
+        val imageUrl: String?
+    )
+
+    /**
+     * Constructs a notification for the timer push template
+     *
+     * @param context the context
+     * @param template the timer push template
+     * @param trackerActivityClass the tracker
+     * @param broadcastReceiverClass the broadcast receiver
+     * @return the notification builder
+     */
+    @Throws(NotificationConstructionFailedException::class)
+    fun construct(
+        context: Context,
+        template: TimerPushTemplate,
+        trackerActivityClass: Class<out Activity>?,
+        broadcastReceiverClass: Class<out BroadcastReceiver>?
+    ): NotificationCompat.Builder {
+
+        Log.trace(LOG_TAG, TAG, "Building a timer template push notification.")
+
+        val notificationManager =
+            context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
+
+        // create the notification channel if needed
+        val channelIdToUse = notificationManager.createNotificationChannelIfRequired(context, template)
+
+        // check if the template is expired
+        val isExpired = template.isExpired()
+        val (smallLayout, expandedLayout) = initializeLayouts(context, isExpired)
+
+        // create the notification builder with the common settings applied
+        val notificationBuilder = AEPPushNotificationBuilder.construct(
+            context,
+            template,
+            channelIdToUse,
+            trackerActivityClass,
+            smallLayout,
+            expandedLayout,
+            R.id.basic_expanded_layout
+        )
+
+        // create the timer content according to the expiry time
+        val timerContent = if (isExpired) {
+            TimerContent(template.alternateTitle, template.alternateBody, template.alternateExpandedBody, template.alternateImage)
+        } else {
+            TimerContent(template.title, template.body, template.expandedBodyText, template.imageUrl)
+        }
+
+        // add text to collapsed layout
+        smallLayout.setTextViewText(R.id.notification_title, timerContent.title)
+        smallLayout.setTextViewText(R.id.notification_body, timerContent.body)
+
+        // add text to expanded layout
+        expandedLayout.setTextViewText(R.id.notification_title, timerContent.title)
+        expandedLayout.setTextViewText(R.id.notification_body_expanded, timerContent.expandedBody)
+
+        // download the image for expanded layout
+        val downloadedImageCount = PushTemplateImageUtils.cacheImages(listOf(timerContent.imageUrl))
+        if (downloadedImageCount == 0) {
+            Log.trace(LOG_TAG, TAG, "Timer push template unable to download the image for url : " + timerContent.imageUrl)
```

Comment:
nit: use interpolation for printing the url: 
```kotlin
Log.trace(LOG_TAG, TAG, "Timer push template unable to download the image for url $timerContent.imageUrl)
```

Reply:
Got it